akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -598,69 +587,74 @@ class ShardCoordinator(typeName: String, settings: ClusterShardingSettings,
     case _: CurrentClusterState ⇒
   }
 
+  def activate() = {
+    context.become(active)
+    unstashAll()
+    log.info("Sharding Coordinator was moved to the active state")
+  }
+
+  def update(evt: DomainEvent): Unit = {
+    state = state.updated(evt)
+    sendUpdate(evt)
+  }
+
+  def getState(): Unit =
+    replicator ! Get(CoordinatorStateKey, ReadMajority(waitingForStateTimeout))
+
+  def sendUpdate(evt: DomainEvent) =
+    replicator ! Update(CoordinatorStateKey, LWWRegister(State.empty), WriteMajority(updatingStateTimeout), Some(evt)) { reg ⇒
+      reg.withValue(reg.value.updated(evt))
```

Comment:
shouldn't this be `reg.withValue(state)`?

I think the approach of applying the event again is problematic when you retry the updates. ordering, duplicate operations?